app/connectors/TariffCommoditiesConnector.scala

```diff
@@ -0,0 +1,98 @@
+/*
+ * Copyright 2021 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package connectors
+
+import com.google.inject.Inject
+import config.AppConfig
+import metrics.{ExportsMetrics, MetricIdentifiers}
+import models.responses.TariffCommoditiesResponse
+import play.api.Logger
+import play.api.http.{ContentTypes, HeaderNames, MimeTypes, Status}
+import play.mvc.Http.Status.OK
+import uk.gov.hmrc.http.{HttpClient, _}
+
+import java.net.URL
+import javax.inject.Singleton
+import scala.concurrent.{ExecutionContext, Future}
+
+@Singleton
+class TariffCommoditiesConnector @Inject()(appConfig: AppConfig, httpClient: HttpClient, metrics: ExportsMetrics)(implicit ec: ExecutionContext) {
+
+  private val logger = Logger(this.getClass)
+
+  def getCommodity(commodityCode: String)(implicit hc: HeaderCarrier): Future[String] = {
+    val timer = metrics.startTimer(MetricIdentifiers.tariffCommoditiesMetric)
+    get(commodityCode) map { res =>
+      logger.debug(s"TARIFF_COMMODITIES response is  --> ${res.toString}")
+      res match {
+        case TariffCommoditiesResponse(OK, Some(commodityJson)) =>
+          timer.stop()
+          commodityJson
+        case TariffCommoditiesResponse(status, _) =>
+          timer.stop()
+          throw new InternalServerException(s"TARIFF_COMMODITIES returned [$status]")
+      }
+    }
+  }
+
+  private[connectors] def get(commodityCode: String)(implicit hc: HeaderCarrier): Future[TariffCommoditiesResponse] = {
+    logger.debug(s"TARIFF_COMMODITIES request code is -> $commodityCode")
+    httpClient
+      .GET(commoditiesUrl(commodityCode), headers = headers())
+      .recover {
+        case error: Throwable =>
+          logger.error(s"Error while fetching commodities: ${error.getMessage}")
+          TariffCommoditiesResponse(Status.INTERNAL_SERVER_ERROR, None)
+      }
+  }
+
+  private def commoditiesUrl(commodityCode: String): URL = new URL(s"${appConfig.tariffCommoditiesUri}/id/$commodityCode")
+
+  private def headers(): Seq[(String, String)] = Seq(HeaderNames.ACCEPT -> MimeTypes.JSON, HeaderNames.CONTENT_TYPE -> ContentTypes.JSON)
```

Comment:
Do you really need the header `HeaderNames.CONTENT_TYPE -> ContentTypes.JSON` ? At the end this is a 'GET'.